packages/api/endpoints/reconciliation-reports.js

```diff
@@ -77,35 +88,41 @@ async function getReport(req, res) {
     );
 
     if (Key.endsWith('.json') || Key.endsWith('.csv')) {
-      const reportSize = await getObjectSize({ s3: s3(), bucket: Bucket, key: Key });
+      const reportSize = await getObjectSize({ s3: s3(), bucket: Bucket, key: Key }) ?? 0;
       // estimated payload size, add extra
       const estimatedPayloadSize = presignedS3Url.length + reportSize + 50;
-      if (
-        estimatedPayloadSize
-          > (process.env.maxResponsePayloadSizeBytes || maxResponsePayloadSizeBytes)
-      ) {
+      let maxResponsePayloadSize;
+      if (process.env.maxResponsePayloadSizeBytes) {
+        maxResponsePayloadSize = Number(process.env.maxResponsePayloadSizeBytes)
+      } else {
+        maxResponsePayloadSize = maxResponsePayloadSizeBytes
+      }
+      if (estimatedPayloadSize > maxResponsePayloadSize) {
         res.json({
           presignedS3Url,
           data: `Error: Report ${name} exceeded maximum allowed payload size`,
         });
       } else {
         const file = await getS3Object(Bucket, Key);
         logger.debug(`Sending json file with contentLength ${file.ContentLength}`);
+        if (!file.Body) {
+          return res.boom.badRequest('Report file does not have a body.');
+        }
         const fileBody = await getObjectStreamContents(file.Body);
```

Comment:
without the above check, ts-check complains about `file.Body`
```
Argument of type 'StreamingBlobTypes | undefined' is not assignable to parameter of type 'Readable'.
  Type 'undefined' is not assignable to type 'ReadableBase'.ts(2345)
```
But even adding the above check in, ts-check then complains:
```
Argument of type 'StreamingBlobTypes' is not assignable to parameter of type 'Readable'.
  Type 'ReadableStream<any>' is missing the following properties from type 'ReadableBase': readableAborted, readable, readableDidRead, readableEncoding, and 51 more.ts(2345)
```
I don't feel like I know enough about the types here, so I will look into how to fix this issue, but I'm also open to advice/suggestions.